_P101_TMP117.ino

```diff
@@ -0,0 +1,269 @@
+#ifdef PLUGIN_BUILD_TESTING
+
+//#######################################################################################################
+//######################## Plugin 101 TMP117 I2C high-accuracy Temperature Sensor ############################################
+//#######################################################################################################
+// by tonydee ### 20-03-2020 Microtech (r), see my projects and blog at https://microtechitalia.it
+#include <TMP117.h>
+
+#define PLUGIN_101
+#define PLUGIN_ID_101        101
+#define PLUGIN_NAME_101       "Temperature High Precision - TMP117 [TESTING]"
+#define PLUGIN_VALUENAME1_101 "Temperature"
+
+// ======================================
+// TMP117 I2C high-accuracy Temperature Sensor
+// ======================================
+
+boolean Plugin_101_init = false;
+// Select the correct address setting
+uint8_t ADDR_GND =  0x48;   // 1001000 
+uint8_t ADDR_VCC =  0x49;   // 1001001
+uint8_t ADDR_SDA =  0x4A;   // 1001010
+uint8_t ADDR_SCL =  0x4B;   // 1001011
+uint8_t ADDR =  ADDR_GND;
+uint8_t ConvTime = 0x00;
+uint8_t Averaging = 0x00;
+float offset = 0.0;
+float adjust = 0.5;
+
+TMP117 tmp(ADDR);           // I2C address for the sensor

```

Comment:
By using these global defined objects, you make it impossible to have multiple instances of this plugin active on the same node.
Also these variable names are very generic, so they are very likely to cause build issues some time as they can be used in other parts of the code as well.
Remember, these will be global defined.

Apart from that, they do take resources even when the plugin is not selected or even set enabled.